web/public/js/directives/event/list/item/index.js

```diff
@@ -16,6 +16,22 @@ angular
             cdELI.isEmbedded = embedder.isEmbedded();
             cdELI.event.isPast = eventUtils.isEventInPast(_.last(cdELI.event.dates));
             cdELI.event = eventUtils.getFormattedDates(cdELI.event);
+            if (!cdELI.event.isPast && cdELI.event.type === 'recurring') {
```

Comment:
we still need the formattedDates which are != from the dates
Plus the while loop is there to ensure to stop when we find the first one. The reason is : over a year of repeated event, it can make a difference perfomance-wise than doing it over every date.